ui/src/app/pages/data/criteria-search.tsx

```diff
@@ -126,16 +128,30 @@ export class CriteriaSearch extends React.Component<Props, State>  {
       mode = 'tree';
     }
     this.setState({backMode, hierarchyNode, mode});
-    this.subscription = currentCohortCriteriaStore.subscribe(currentCohortCriteria => {
-      if (source === 'criteria') {
+    if (source === 'conceptSetDetails') {
+      this.getConceptSet();
+    }
+    if (source === 'criteria') {
+      this.subscription = currentCohortCriteriaStore.subscribe(currentCohortCriteria => {
         this.setState({selectedCriteriaList: currentCohortCriteria});
-      }
-    });
-    this.subscription.add(currentConceptStore.subscribe(currentConcepts => {
-      if (source === 'concept') {
+      });
+    } else {
+      this.subscription = currentConceptStore.subscribe(currentConcepts => {
         this.setState({selectedCriteriaList: currentConcepts});
-      }
-    }));
+      });
+    }
+  }
+
+  async getConceptSet() {
+    const {urlParams: {ns, wsid, csid}} = this.props;
+    try {
+      const resp = await conceptSetsApi().getConceptSet(ns, wsid, csid);
```

Comment:
I noticed this is called twice because it's getting called in `concept-set-details` component first, which stores the concept set in `currentConceptSetStore`. Maybe check if the the concept set in the store matches the csid in props before calling the api to avoid duplicate calls.

Reply:
Done